File: apps/android/core/model/src/main/kotlin/com/scribblefit/core/model/ExerciseHistorySession.kt
package com.scribblefit.core.model

import java.util.UUID

/**
 * Domain model representing a single historical session for an exercise.
 */
data class ExerciseHistorySession(
    val exercise: Exercise,
    val totalVolume: Float,
    val maxWeight: Float,
    val summary: String,
    val isPersonalBest: Boolean,
    val scribbleId: Long
) {
    val date: Long get() = exercise.createdAt
}

File: apps/ios/ScribbleFit/LocalPackages/Core/CoreModel/Sources/CoreModel/ExerciseHistorySession.swift
import Foundation

public struct ExerciseHistorySession: Identifiable, Codable, Sendable, Equatable {
    public let id: UUID
    public let exercise: Exercise
    public let totalVolume: Float
    public let maxWeight: Float
    public let summary: String
    public let isPersonalBest: Bool
    public let scribbleId: UUID
    
    public var date: Date { exercise.createdAt }
    
    public init(
        id: UUID = UUID(),
        exercise: Exercise,
        totalVolume: Float,
        maxWeight: Float,
        summary: String,
        isPersonalBest: Bool,
        scribbleId: UUID
    ) {
        self.id = id
        self.exercise = exercise
        self.totalVolume = totalVolume
        self.maxWeight = maxWeight
        self.summary = summary
        self.isPersonalBest = isPersonalBest
        self.scribbleId = scribbleId
    }
}
